sdk/core/azure-core/inc/azure/core/internal/tracing/service_tracing.hpp

```diff
@@ -211,12 +208,15 @@ namespace Azure { namespace Core { namespace Tracing { namespace _internal {
         Azure::Core::Tracing::_internal::SpanKind const& spanKind,
         Azure::Core::Context const& clientContext);
 
-    static ContextAndSpan CreateSpanFromContext(
+    ContextAndSpan CreateSpan(
         std::string const& spanName,
-        Azure::Core::Tracing::_internal::SpanKind const& spanKind,
+        Azure::Core::Tracing::_internal::CreateSpanOptions& spanOptions,
         Azure::Core::Context const& clientContext);
 
     std::unique_ptr<Azure::Core::Tracing::_internal::AttributeSet> CreateAttributeSet();
+
+    static DiagnosticTracingFactory* DiagnosticFactoryFromContext(
```

Comment:
Would be better to return this is a std::unique_ptr so lifetime ownership is clear.

Reply:
Let me see what I can do.